cooked-validators/src/Cooked/Tweak/TamperDatum.hs

```diff
@@ -46,4 +47,19 @@ tamperDatumTweak change = do
   addLabelTweak TamperDatumLbl
   return beforeModification
 
+-- | Similar as 'tamperDatumTweak' with a predicate and a transformation
+tamperDatumPredTweak ::
+  forall a m.
+  ( MonadTweak m,
+    Show a,
+    Pretty a,
+    Pl.ToData a,
+    Pl.FromData a,
+    Typeable a
+  ) =>
+  (a -> a) ->
+  (a -> Bool) ->
+  m [a]
+tamperDatumPredTweak = (tamperDatumTweak .) . ifMaybe
```

Comment:
The same answer applies. In practice, I see myself using a lot more this version that the one with `Maybe` because I think from an engineering perspective, it's often more rigorous to split the two questions:
- which elements should be modified?
- which modification should be applied on them?
Rather than having them both in the same transformation.
I like the version with Maybe though because it allows for the selection you implemented and it alone encompasses all we need to know to apply what needs to be applied on the right element, i just think that having an additional layer with the predicate is more natural and will fit nicely into our usual use cases in audits.